.circleci/config.yml

```diff
@@ -410,6 +410,29 @@ commands:
       # Allow this to fail, but it is useful for debugging.
       - run: sh -c "pip check || true"
 
+  setup-docker-env:
+    description: Set up and create docker env.
+    parameters:
+      image:
+        type: string
+      image-name:
+        type: string
+        default: "test_container"
+      bind-mount:
+        type: string
+        default: "$PWD"
+      extras-requires:
+        type: string
+        default: ""
+    steps:
+      - run: docker run -v <<parameters.bind-mount>>:/usr/src/project --gpus=all -it -d --name <<parameters.image-name>> --workdir /usr/src/project <<parameters.image>>
+      - run:
+          name: Install <<parameters.extras-requires>>
+          command: |
+            if [ -n "<<parameters.extras-requires>>" ]; then
+              docker exec <<parameters.image-name>> tools/scripts/retry.sh pip install --progress-bar off <<parameters.extras-requires>>
```

Comment:
I would _love_ to see us get rid of all test-time pip installs...  But I do think it is wise to do that in a separate PR.